
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>Taxonomic classification of reads &#8212; MOSHPIT plugin suite</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles.css?v=3437f865" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'chapters/03_taxonomic_classification/reads';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Taxonomic classification of MAGs" href="mags.html" />
    <link rel="prev" title="Taxonomic classification" href="intro.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo_light.svg" class="logo__image only-light" alt="MOSHPIT plugin suite - Home"/>
    <img src="../../_static/logo_dark.svg" class="logo__image only-dark pst-js-only" alt="MOSHPIT plugin suite - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../00_setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../00_data_retrieval.html">Data retrieval</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../01_filtering/intro.html">Quality control</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../01_filtering/quality-filtering.html">Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="../01_filtering/host-filtering.html">Host read removal</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../02_mag_reconstruction/intro.html">MAG recovery</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../02_mag_reconstruction/reconstruction.html">MAG reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="../02_mag_reconstruction/dereplication.html">Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="../02_mag_reconstruction/abundance.html">Abundance estimation</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="intro.html">Taxonomic classification</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="mags.html">MAGs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../04_functional_annotation/intro.html">Functional annotation</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../04_functional_annotation/mags.html">Dereplicated MAGs</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/chapters/03_taxonomic_classification/reads.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>Taxonomic classification of reads</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#approach-1-kraken-2">Approach 1: Kraken 2</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#approach-2-kaiju">Approach 2: Kaiju</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="taxonomic-classification-of-reads">
<span id="kraken-reads"></span><h1>Taxonomic classification of reads<a class="headerlink" href="#taxonomic-classification-of-reads" title="Link to this heading">#</a></h1>
<p>In this section we will focus on the taxonomic classification of shotgun metagenomic reads using two different tools: Kraken 2 and Kaiju.
We will use the data obtained in the <a class="reference internal" href="../00_data_retrieval.html"><span class="std std-doc">data retrieval section</span></a>.</p>
<section id="approach-1-kraken-2">
<h2>Approach 1: Kraken 2<a class="headerlink" href="#approach-1-kraken-2" title="Link to this heading">#</a></h2>
<p>Before we can use Kraken 2, we need to build or download a database. We will use the <code class="docutils literal notranslate"><span class="pre">build-kraken-db</span></code> action to fetch the PlusPF database
from <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">here</a> - this database covers RefSeq sequences for archaea, bacteria, viral, plasmid,
human, UniVec_Core, protozoa and fungi.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">build</span><span class="o">-</span><span class="n">kraken</span><span class="o">-</span><span class="n">db</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">collection</span> <span class="n">pluspf</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">kraken2</span><span class="o">-</span><span class="n">database</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken2_db</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">bracken</span><span class="o">-</span><span class="n">database</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_db</span> \
</pre></div>
</div>
</div>
</div>
<p>We can now use the <code class="docutils literal notranslate"><span class="pre">classify-kraken2</span></code> command to run Kraken2 using the paired-end reads as a query and the PlusPF database retrieved in the previous step:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">classify</span><span class="o">-</span><span class="n">kraken2</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">seqs</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">reads_filtered</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">kraken2</span><span class="o">-</span><span class="n">db</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken2_db</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threads</span> <span class="mi">72</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">confidence</span> <span class="mf">0.5</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">memory</span><span class="o">-</span><span class="n">mapping</span> <span class="kc">False</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">report</span><span class="o">-</span><span class="n">minimizer</span><span class="o">-</span><span class="n">data</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">reports</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_reports_reads</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">hits</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_hits_reads</span>
  <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
</div>
</div>
<div class="admonition seealso">
<p class="admonition-title">See also</p>
<p><a class="reference external" href="https://ccb.jhu.edu/software/bracken/">Bracken</a> is a related tool that additionally estimates relative abundances of species or genera to adjust for
the genome size the organisms from which each read originated. In order to use this tool we need the Bracken database that was fetched in the first step.</p>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">estimate</span><span class="o">-</span><span class="n">bracken</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">kraken</span><span class="o">-</span><span class="n">reports</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_reports_reads</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">bracken</span><span class="o">-</span><span class="n">db</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_db</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threshold</span> <span class="mi">5</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">read</span><span class="o">-</span><span class="nb">len</span> <span class="mi">150</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_taxonomy</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_ft</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">reports</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_reports</span>
</pre></div>
</div>
</div>
</div>
<p>To remove the unclassified read fraction we can use the <code class="docutils literal notranslate"><span class="pre">filter-table</span></code> action from the <code class="docutils literal notranslate"><span class="pre">q2-taxa</span></code> QIIME 2 plugin:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">taxa</span> <span class="nb">filter</span><span class="o">-</span><span class="n">table</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_ft</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_taxonomy</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">exclude</span> <span class="n">Unclassified</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">bracken_ft_filtered</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="approach-2-kaiju">
<h2>Approach 2: Kaiju<a class="headerlink" href="#approach-2-kaiju" title="Link to this heading">#</a></h2>
<p>Similarly to Kraken 2, Kaiju requires a reference database to perform taxonomic classification. We will use the <code class="docutils literal notranslate"><span class="pre">fetch-kaiju-db</span></code>
action to download the <a class="reference external" href="https://bioinformatics-centre.github.io/kaiju/downloads.html">nr_euk</a> database that includes both
prokaryotes and eukaryotes (more info on the taxa <a class="reference external" href="https://github.com/bioinformatics-centre/kaiju/blob/master/util/kaiju-taxonlistEuk.tsv">here</a>).</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">fetch</span><span class="o">-</span><span class="n">kaiju</span><span class="o">-</span><span class="n">db</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">database</span><span class="o">-</span><span class="nb">type</span> <span class="n">nr_euk</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">database</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_nr_euk</span>
</pre></div>
</div>
</div>
</div>
<p>We run Kaiju with the confidence of 0.1 using the paired-end reads as a query and the database artifact that was generated in the previous step:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">classify</span><span class="o">-</span><span class="n">kaiju</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">seqs</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">reads_paired</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">db</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_nr_euk</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">z</span> <span class="mi">16</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">c</span> <span class="mf">0.1</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_taxonomy</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">abundances</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_ft</span>
</pre></div>
</div>
</div>
</div>
<p>Finally, we filter the table to remove the unclassified reads:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">taxa</span> <span class="nb">filter</span><span class="o">-</span><span class="n">table</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_ft</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_taxonomy</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">exclude</span> <span class="n">unclassified</span><span class="p">,</span><span class="n">belong</span><span class="p">,</span><span class="n">cannot</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_ft_filtered</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="visualization">
<h2>Visualization<a class="headerlink" href="#visualization" title="Link to this heading">#</a></h2>
<p>You can try to generate a taxa bar plot with either of these results now! We will continue with the Kaiju results - to
generate a taxa bar plot, you can run:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">taxa</span> <span class="n">barplot</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_ft_filtered</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kaiju_taxonomy</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="o">./</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="o">./</span><span class="n">results</span><span class="o">/</span><span class="n">kaiju_barplot</span><span class="o">.</span><span class="n">qzv</span>
</pre></div>
</div>
</div>
</div>
<p>Your visualization should look similar to <a class="reference external" href="https://view.qiime2.org/visualization/?src=https://raw.githubusercontent.com/bokulich-lab/moshpit-docs/main/moshpit_docs/data/kaiju-filtered.qzv">this one</a>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./chapters/03_taxonomic_classification"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="intro.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Taxonomic classification</p>
      </div>
    </a>
    <a class="right-next"
       href="mags.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Taxonomic classification of MAGs</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#approach-1-kraken-2">Approach 1: Kraken 2</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#approach-2-kaiju">Approach 2: Kaiju</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#visualization">Visualization</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By The Bokulich Lab
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>