
<!DOCTYPE html>


<html lang="en" data-content_root="../../" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="viewport" content="width=device-width, initial-scale=1" />

    <title>MAG abundance estimation &#8212; MOSHPIT plugin suite</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "";
  </script>
  <!-- 
    this give us a css class that will be invisible only if js is disabled 
  -->
  <noscript>
    <style>
      .pst-js-only { display: none !important; }

    </style>
  </noscript>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="../../_static/styles/theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />
<link href="../../_static/styles/pydata-sphinx-theme.css?digest=26a4bc78f4c0ddb94549" rel="stylesheet" />

    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css?v=fa44fd50" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles/sphinx-book-theme.css?v=a3416100" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css?v=13237357" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css?v=76b2166b" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css?v=4fa983c6" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-design.min.css?v=95c83b7e" />
    <link rel="stylesheet" type="text/css" href="../../_static/styles.css?v=3437f865" />
  
  <!-- So that users can add custom icons -->
  <script src="../../_static/scripts/fontawesome.js?digest=26a4bc78f4c0ddb94549"></script>
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549" />
<link rel="preload" as="script" href="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549" />

    <script src="../../_static/documentation_options.js?v=9eb32ce0"></script>
    <script src="../../_static/doctools.js?v=9a2dae69"></script>
    <script src="../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../_static/clipboard.min.js?v=a7894cd8"></script>
    <script src="../../_static/copybutton.js?v=f281be69"></script>
    <script src="../../_static/scripts/sphinx-book-theme.js?v=887ef09a"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../../_static/togglebutton.js?v=4a39c7ea"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script src="../../_static/design-tabs.js?v=f930bc37"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script async="async" src="../../_static/sphinx-thebe.js?v=c100c467"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown';</script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"; const thebe_selector = ".thebe,.cell"; const thebe_selector_input = "pre"; const thebe_selector_output = ".output, .cell_output"</script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'chapters/02_mag_reconstruction/abundance';</script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Taxonomic classification" href="../03_taxonomic_classification/intro.html" />
    <link rel="prev" title="MAG set dereplication" href="dereplication.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  <meta name="docsearch:version" content="" />
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <div id="pst-skip-link" class="skip-link d-print-none"><a href="#main-content">Skip to main content</a></div>
  
  <div id="pst-scroll-pixel-helper"></div>
  
  <button type="button" class="btn rounded-pill" id="pst-back-to-top">
    <i class="fa-solid fa-arrow-up"></i>Back to top</button>

  
  <dialog id="pst-search-dialog">
    
<form class="bd-search d-flex align-items-center"
      action="../../search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form>
  </dialog>

  <div class="pst-async-banner-revealer d-none">
  <aside id="bd-header-version-warning" class="d-none d-print-none" aria-label="Version warning"></aside>
</div>

  
    <header class="bd-header navbar navbar-expand-lg bd-navbar d-print-none">
    </header>
  

  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      
      
      <dialog id="pst-primary-sidebar-modal"></dialog>
      <div id="pst-primary-sidebar" class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">

  
    
  

<a class="navbar-brand logo" href="../../intro.html">
  
  
  
  
  
    
    
      
    
    
    <img src="../../_static/logo_light.svg" class="logo__image only-light" alt="MOSHPIT plugin suite - Home"/>
    <img src="../../_static/logo_dark.svg" class="logo__image only-dark pst-js-only" alt="MOSHPIT plugin suite - Home"/>
  
  
</a></div>
        <div class="sidebar-primary-item">

<button class="btn search-button-field search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
 <i class="fa-solid fa-magnifying-glass"></i>
 <span class="search-button__default-text">Search</span>
 <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd class="kbd-shortcut__modifier">K</kbd></span>
</button></div>
        <div class="sidebar-primary-item"><nav class="bd-links bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <ul class="current nav bd-sidenav">
<li class="toctree-l1"><a class="reference internal" href="../00_setup.html">Setup</a></li>
<li class="toctree-l1"><a class="reference internal" href="../00_data_retrieval.html">Data retrieval</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../01_filtering/intro.html">Quality control</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../01_filtering/quality-filtering.html">Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="../01_filtering/host-filtering.html">Host read removal</a></li>
</ul>
</details></li>
<li class="toctree-l1 current active has-children"><a class="reference internal" href="intro.html">MAG recovery</a><details open="open"><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="reconstruction.html">MAG reconstruction</a></li>
<li class="toctree-l2"><a class="reference internal" href="dereplication.html">Dereplication</a></li>
<li class="toctree-l2 current active"><a class="current reference internal" href="#">Abundance estimation</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../03_taxonomic_classification/intro.html">Taxonomic classification</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../03_taxonomic_classification/reads.html">Reads</a></li>
<li class="toctree-l2"><a class="reference internal" href="../03_taxonomic_classification/mags.html">MAGs</a></li>
</ul>
</details></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="../04_functional_annotation/intro.html">Functional annotation</a><details><summary><span class="toctree-toggle" role="presentation"><i class="fa-solid fa-chevron-down"></i></span></summary><ul>
<li class="toctree-l2"><a class="reference internal" href="../04_functional_annotation/mags.html">Dereplicated MAGs</a></li>
</ul>
</details></li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main" role="main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article d-print-none">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><button class="sidebar-toggle primary-toggle btn btn-sm" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</button></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="../../_sources/chapters/02_mag_reconstruction/abundance.md" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.md</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>



<button class="btn btn-sm nav-link pst-navbar-icon theme-switch-button pst-js-only" aria-label="Color mode" data-bs-title="Color mode"  data-bs-placement="bottom" data-bs-toggle="tooltip">
  <i class="theme-switch fa-solid fa-sun                fa-lg" data-mode="light" title="Light"></i>
  <i class="theme-switch fa-solid fa-moon               fa-lg" data-mode="dark"  title="Dark"></i>
  <i class="theme-switch fa-solid fa-circle-half-stroke fa-lg" data-mode="auto"  title="System Settings"></i>
</button>


<button class="btn btn-sm pst-navbar-icon search-button search-button__button pst-js-only" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass fa-lg"></i>
</button>
<button class="sidebar-toggle secondary-toggle btn btn-sm" title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</button>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>MAG abundance estimation</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#get-mag-lengths">Get MAG lengths</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#index-dereplicated-mags">Index dereplicated MAGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#map-reads-to-dereplicated-mags">Map reads to dereplicated MAGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#estimate-mag-abundance">Estimate MAG abundance</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#let-s-have-a-look-at-our-estimated-mag-abundance">Let’s have a look at our estimated MAG abundance!</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article">
                  
  <section class="tex2jax_ignore mathjax_ignore" id="mag-abundance-estimation">
<h1>MAG abundance estimation<a class="headerlink" href="#mag-abundance-estimation" title="Link to this heading">#</a></h1>
<p>Once we recover MAGs from metagenomic data, we may be interested in estimating their abundance in the samples. We can do
it by mapping the original reads to the dereplicated MAGs and calculating the abundance based on the read mapping results.
There are a couple of ways to estimate MAG abundance, such as RPKM (Reads Per Kilobase per Million mapped reads) and TPM
(Transcripts Per Million). Here we will use TPM to estimate the abundance of each MAG in all samples.</p>
<section id="get-mag-lengths">
<h2>Get MAG lengths<a class="headerlink" href="#get-mag-lengths" title="Link to this heading">#</a></h2>
<p>This step calculates the lengths of each dereplicated MAG, which will be used in the next step to estimate abundance.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">get</span><span class="o">-</span><span class="n">feature</span><span class="o">-</span><span class="n">lengths</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">features</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep</span> \              
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">lengths</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_length</span> \ 
    <span class="o">--</span><span class="n">verbose</span>                         
</pre></div>
</div>
</div>
</div>
</section>
<section id="index-dereplicated-mags">
<h2>Index dereplicated MAGs<a class="headerlink" href="#index-dereplicated-mags" title="Link to this heading">#</a></h2>
<p>This step indexes the dereplicated MAGs for read mapping. The index is necessary to efficiently map the input reads back to the MAGs.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">assembly</span> <span class="n">index</span><span class="o">-</span><span class="n">derep</span><span class="o">-</span><span class="n">mags</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">mags</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep</span> \                  
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threads</span> <span class="mi">8</span> \  
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">seed</span> <span class="mi">100</span> \                                   
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">index</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_index</span> \
    <span class="o">--</span><span class="n">verbose</span>                            
</pre></div>
</div>
</div>
</div>
</section>
<section id="map-reads-to-dereplicated-mags">
<h2>Map reads to dereplicated MAGs<a class="headerlink" href="#map-reads-to-dereplicated-mags" title="Link to this heading">#</a></h2>
<p>In this step, we map the input paired-end reads back to the dereplicated MAGs. This helps in calculating the abundance
of each MAG in the sample.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">assembly</span> <span class="nb">map</span><span class="o">-</span><span class="n">reads</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">index</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_index</span> \                            
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">reads</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">reads_filtered</span> \   
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threads</span> <span class="mi">8</span> \  
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">seed</span> <span class="mi">100</span> \                  
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">alignment</span><span class="o">-</span><span class="nb">map</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">reads_to_derep_mags</span> \
    <span class="o">--</span><span class="n">verbose</span>            
</pre></div>
</div>
</div>
</div>
</section>
<section id="estimate-mag-abundance">
<h2>Estimate MAG abundance<a class="headerlink" href="#estimate-mag-abundance" title="Link to this heading">#</a></h2>
<p>This step estimates the abundance of each MAG in the sample based on the read mapping results.</p>
<ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">metric</span></code> : currently, we support RPKM and TPM</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">min-mapq</span></code> : indicates the minimum required read mapping quality - for Bowtie2, 42 will allow only perfect matches to be retained
<code class="docutils literal notranslate"><span class="pre">-</span> <span class="pre">min-base-quality</span></code> : only keep alignments with this minimal Phred quality score</p></li>
</ul>
<p>For more options, see –help.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">estimate</span><span class="o">-</span><span class="n">mag</span><span class="o">-</span><span class="n">abundance</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">mag</span><span class="o">-</span><span class="n">lengths</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_length</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">maps</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">reads_to_derep_mags</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threads</span> <span class="mi">10</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">metric</span> <span class="n">tpm</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="nb">min</span><span class="o">-</span><span class="n">mapq</span> <span class="mi">42</span> \
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">abundances</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_ft</span> \
    <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="let-s-have-a-look-at-our-estimated-mag-abundance">
<h2>Let’s have a look at our estimated MAG abundance!<a class="headerlink" href="#let-s-have-a-look-at-our-estimated-mag-abundance" title="Link to this heading">#</a></h2>
<p>First we will use Kraken 2 to classify provided MAGs into taxonomic groups.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Refer to <a class="reference internal" href="../03_taxonomic_classification/reads.html#kraken-reads"><span class="std std-ref">Taxonomic classification of reads</span></a> section for more details on taxonomic classification with Kraken 2.</p>
</div>
<p>The database used here is the <code class="docutils literal notranslate"><span class="pre">PlusPF</span></code> database, defined <a class="reference external" href="https://benlangmead.github.io/aws-indexes/k2">here</a>.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">classify</span><span class="o">-</span><span class="n">kraken2</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">seqs</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">kraken2</span><span class="o">-</span><span class="n">db</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken2_db</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">threads</span> <span class="mi">40</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">confidence</span> <span class="mf">0.5</span> \
    <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">report</span><span class="o">-</span><span class="n">minimizer</span><span class="o">-</span><span class="n">data</span> \
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">reports</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_reports_mags_derep</span> \
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">hits</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_hits_mags_derep</span> \
    <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
</div>
</div>
<p>Then we will convert a Kraken 2 report into a  generic taxonomy artifact for downstream analyses.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">moshpit</span> <span class="n">kraken2</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">mag</span><span class="o">-</span><span class="n">features</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">reports</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_reports_mags_derep</span>  \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">hits</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">kraken_hits_mags_derep</span>  \
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_taxonomy</span> \
    <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
</div>
</div>
<p>Now we are ready to generate a taxa bar plot.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-python notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">taxa</span> <span class="n">barplot</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">table</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_ft</span> \
    <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">./</span><span class="n">cache</span><span class="p">:</span><span class="n">mags_derep_taxonomy</span> \
    <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="o">./</span><span class="n">cocoa</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
    <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="o">./</span><span class="n">results</span><span class="o">/</span><span class="n">mags</span><span class="o">-</span><span class="n">derep</span><span class="o">-</span><span class="n">taxa</span><span class="o">-</span><span class="n">bar</span><span class="o">-</span><span class="n">plot</span><span class="o">.</span><span class="n">qzv</span> \
    <span class="o">--</span><span class="n">verbose</span>
</pre></div>
</div>
</div>
</div>
<p>Your visualization should look similar to <a class="reference external" href="https://view.qiime2.org/visualization/?src=https://raw.githubusercontent.com/bokulich-lab/moshpit-docs/main/moshpit_docs/data/mags-derep-taxa-bar-plot.qzv">this one</a>.</p>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            name: "python3",
            path: "./chapters/02_mag_reconstruction"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

                </article>
              

              
              
              
              
                <footer class="prev-next-footer d-print-none">
                  
<div class="prev-next-area">
    <a class="left-prev"
       href="dereplication.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">MAG set dereplication</p>
      </div>
    </a>
    <a class="right-next"
       href="../03_taxonomic_classification/intro.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Taxonomic classification</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <dialog id="pst-secondary-sidebar-modal"></dialog>
                <div id="pst-secondary-sidebar" class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">


  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#get-mag-lengths">Get MAG lengths</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#index-dereplicated-mags">Index dereplicated MAGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#map-reads-to-dereplicated-mags">Map reads to dereplicated MAGs</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#estimate-mag-abundance">Estimate MAG abundance</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#let-s-have-a-look-at-our-estimated-mag-abundance">Let’s have a look at our estimated MAG abundance!</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By The Bokulich Lab
</p>

  </div>
  
  <div class="footer-item">
    

  <p class="copyright">
    
      © Copyright 2023.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script defer src="../../_static/scripts/bootstrap.js?digest=26a4bc78f4c0ddb94549"></script>
<script defer src="../../_static/scripts/pydata-sphinx-theme.js?digest=26a4bc78f4c0ddb94549"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>